gson2/src/main/java/org/jdbi/v3/gson2/GsonJsonMapper.java

```diff
@@ -24,7 +26,7 @@
 
 class GsonJsonMapper implements JsonMapper {
     @Override
-    public TypedJsonMapper forType(Type type, ConfigRegistry config) {
```

Comment:
these are all non-backwards compatible changes which would require us to bump major (!) version. They can not be applied in the current shape.

You need to retain backwards compatibility if you want to add additional methods. 

See https://github.com/jdbi/jdbi/actions/runs/16097134400/job/45421456804